### ISSUE TITLE
What kind of weight do we need to use to load llama2?

### ISSUE THREAD

**Issue by Yingshu97** [Vision-CAIR/MiniGPT-4#488]
In the codebase, this project loads llama2 as torch.float16. But I found that the official doc said the llama2 was trained on bfloat16.

llama_model = LlamaForCausalLM.from_pretrained(
                llama_model_path,
                torch_dtype=torch.float16,
            )

When I tried to use Lora to fine-tune Llama. I found that if I load llama as float16, it will give nan loss. But if I load llama as bfloat16, everything goes well.

I don't know the exact reason. Could some experts give me some ideas?
